mteb/models/bmretriever_models.py

```diff
@@ -0,0 +1,190 @@
+from __future__ import annotations
+
+from functools import partial
+from typing import Any, Callable
+
+import torch
+from sentence_transformers import SentenceTransformer
+from sentence_transformers.models import Pooling, Transformer
+
+from mteb.encoder_interface import PromptType
+from mteb.model_meta import ModelMeta
+from mteb.models.instruct_wrapper import InstructSentenceTransformerWrapper
+
+
+def instruction_template(
+    instruction: str, prompt_type: PromptType | None = None
+) -> str:
+    return (
+        f"{instruction}\nQuery: "
+        if (prompt_type is None or prompt_type == PromptType.query) and instruction
+        else "Represent this passage\npassage: "
+    )
+
+
+class BMRetrieverWrapper(InstructSentenceTransformerWrapper):
+    def __init__(
+        self,
+        model_name: str,
+        instruction_template: Callable[[str, PromptType | None], str] | None = None,
+        max_seq_length: int | None = None,
+        apply_instruction_to_passages: bool = True,
+        padding_side: str | None = None,
+        add_eos_token: bool = False,
+        prompts_dict: dict[str, str] | None = None,
+        **kwargs: Any,
+    ):
+        self.model_name = model_name
+        self.instruction_template = instruction_template
+        self.apply_instruction_to_passages = apply_instruction_to_passages
+        self.add_eos_token = add_eos_token
+        self.prompts_dict = prompts_dict
+
+        transformer = Transformer(
+            model_name,
+            max_seq_length=max_seq_length,
+            **kwargs,
+        )
+        pooling = Pooling(
+            transformer.get_word_embedding_dimension(), pooling_mode="lasttoken"
+        )
+        self.model = SentenceTransformer(modules=[transformer, pooling])
+
+        if max_seq_length is not None:
+            self.model.max_seq_length = max_seq_length
+
+        if padding_side is not None:
+            self.model.tokenizer.padding_side = padding_side
+
+
+BMRETRIEVER_TRAINING_DATA = {
+    "MedRAG/pubmed": ["train"],
+    "mteb/raw_arxiv": ["train"],
+    "medalpa/medical_meadow_cord19": ["train"],
+    "MedRAG/textbooks": ["train"],
+    "MedRAG/statpearls": ["train"],
+    "KushT/LitCovid_BioCreative": ["train"],
+    "S2ORC": ["train"],
+    "MSMARCO": ["train"],
+    "BMRetriever/biomed_retrieval_dataset": ["train"],
+}
```

Comment:
These have to be MTEB datasets, where the key is the task name. I would just comment out the rest:
```suggestion
BMRETRIEVER_TRAINING_DATA = {
    "MedRAG/pubmed": ["train"],
    "mteb/raw_arxiv": ["train"],
    "medalpa/medical_meadow_cord19": ["train"],
    "MedRAG/textbooks": ["train"],
    "MedRAG/statpearls": ["train"],
    "KushT/LitCovid_BioCreative": ["train"],
    "S2ORC": ["train"],
    "MSMARCO": ["train"],
    # non-mteb
    # "BMRetriever/biomed_retrieval_dataset": ["train"],
}
```

Reply:
So, do you mean I should change it like this, to only keep MSMARCO?

```python
BMRETRIEVER_TRAINING_DATA = {
    "MSMARCO": ["train"],
    # non-mteb
    # "mteb/raw_arxiv": ["train"],
    # "MedRAG/pubmed": ["train"],
    # "medalpa/medical_meadow_cord19": ["train"],
    # "MedRAG/textbooks": ["train"],
    # "MedRAG/statpearls": ["train"],
    # "KushT/LitCovid_BioCreative": ["train"],
    # "S2ORC": ["train"],
    # "BMRetriever/biomed_retrieval_dataset": ["train"],
}
```